### PR TITLE
feat(node:process): add polyfill for `process.hrtime`

### PR DESCRIPTION
**Pull request by alan-agius4**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

`process.hrtime` is currently not implemented.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/node/process/_process.ts [hunk 1]
```diff
@@ -226,3 +226,12 @@ process.chdir = function (dir) {
 process.umask = function () {
   return 0;
 };
+
+// https://nodejs.org/api/process.html#processhrtime
+process.hrtime = function () {
+  return [0, 0];
+};
```
Comment (IgorMinar) [inline src/runtime/node/process/_process.ts:233]: hey @alan-agius4, I think you actually want something closer to this:

```suggestion
process.hrtime = function (startTime) {
  const now = Date.now();
  // millis to seconds
  const seconds = Math.trunc(now / 1_000);
  // convert millis to nanos
  const nanos = (now % 1_000) * 1_000_000;
  
  if (startTime) {
    let diffSeconds = seconds - startTime[0];
    let diffNanos = nanos - startTime[0];
    
    if (diffNanos < 0) {
      diffSeconds = diffSeconds - 1;
      diffNanos = 1_000_000_000 + diffNanos;
    }
    return [diffSeconds, diffNanos]
  }
  
  return [seconds, nanos];
};
```

**Review by IgorMinar — COMMENTED** [review]
This does look like a good change to me though, especially if the hrtime polyfill is tweaked.

**Review by pi0 — APPROVED** [review]
Thank you both ❤️

File: src/runtime/node/process/_process.ts (hunk 1 above)
Comment (alan-agius4) [reply src/runtime/node/process/_process.ts]: Thanks @IgorMinar 😀

**Pull request merged**